lib/Doctrine/ODM/MongoDB/Mapping/MappingException.php

```diff
@@ -291,4 +291,9 @@ public static function nonBackedEnumMapped(string $className, string $fieldName,
             $fieldName,
         ));
     }
+
+    public static function emptySearchIndexDefinition(string $className, string $indexName): self
+    {
+        return new self(sprintf('%s search index "%s" must be dynamic or specify a field mapping.', $className, $indexName));
```

Comment:
```suggestion
        return new self(sprintf('%s search index "%s" must be dynamic or specify a field mapping', $className, $indexName));
```

I don't think periods are used for single-sentence exception messages. Disregard if I'm mistaken.